soc/riscv/riscv-privilege/Kconfig

```diff
@@ -18,5 +18,15 @@ config RISCV_HAS_PLIC
 	help
 	  Does the SOC provide support for a Platform Level Interrupt Controller
 
+config SOC_CUSTOM_GP_INIT
```

Comment:
looks like we do not need this config.

Telink B91 ``start`` calls ``__start`` from ``common/vector.S`` and here we can init GP as well.
Is there any specific reasons why we should init GP immediately in ``start.S`` but not in ``vector.S``?

If no, I propose to remove ``SOC_CUSTOM_GP_INIT`` config.
Tested telink_b91 solution with GP initialization in ``vector.S`` and no issues detected.

Reply:
I agree. SoC specific entry points are (usually) written in ASM, which makes it easy to add a local `.option norelax`.

Reply:
@yurvyn
In general, load, store, and branch instructions are possible to be optimized to GP-relative addressing (if relaxation isn't disabled.), so we recommand to initialize GP at program start to avoid these issues (or you should be familar with this optimization about GP.)

@yurvyn @henrikbrixandersen 
Yes, another approach is adding `.norelax` option to all code before `__start`. It's almost the ASM code before `__start` because $SP register isn't initialized.

I'm OK with `.norelax` option approach and remove `CONFIG_SOC_CUSTOM_GP`. I can update it tomorrow.

Reply:
If we are to get it into v2.7.x, it likely needs to be done by today. Otherwise it will have to wait.

Reply:
@henrikbrixandersen Thanks for remind. I'm going to fix it soon.

Reply:
Change to `.norelax` option approach and remove `CONFIG_SOC_CUSTOM_GP`